src/main.ts

```diff
@@ -1595,18 +1607,18 @@ export default class WhatsApp {
     return async (e: WebSocket.OpenEvent) => {
       if (
         !restoreSession ||
-        (restoreSession && !(await doesFileExist(this.keysPath!)))
+        (restoreSession && !(await doesFileExist(this.parameters.keysPath!)))
       ) {
         this.clientId = crypto.randomBytes(16).toString("base64");
       } else {
         await this.getKeys();
       }
 
       e.target.send(
-        `${loginMsgId},["admin","init",[0,4,2080],["WhatsApp forwarder","WhatsAppForwarder"],"${this.clientId}",true]`
+        `${loginMsgId},["admin","init",[0,4,2080],["${this.parameters.clientName}","${this.parameters.clientShortName}"],"${this.clientId}",true]`
       );
 
-      if (restoreSession && (await doesFileExist(this.keysPath!))) {
+      if (restoreSession && (await doesFileExist(this.parameters.keysPath!))) {
```

Comment:
```suggestion
      if (restoreSession && (await doesFileExist(this.opts.keysPath!))) {
```
Update property again